src/main/java/org/kohsuke/github/GitHubBuilder.java

```diff
@@ -311,19 +314,39 @@ public GitHubBuilder withConnector(HttpConnector connector) {
     }
 
     /**
-     * With rate limit handler git hub builder.
+     * Adds a {@link RateLimitHandler} to this {@link GitHubBuilder}.
+     * <p>
+     * GitHub allots a certain number of requests to each user or application per period of time (usually per hour). The
+     * number of requests remaining is returned in the response header and can also be requested using
+     * {@link GitHub#getRateLimit()}. This requests per interval is referred to as the "rate limit".
+     * </p>
+     * <p>
+     * When the remaining number of requests reaches zero, the next request will return an error. If this happens,
+     * {@link RateLimitHandler#onError(IOException, HttpURLConnection)} will be called.
+     * </p>
+     * <p>
+     * NOTE: GitHub treats clients that exceed their rate limit very harshly. If possible, clients should avoid
```

Comment:
Is there an official source out there that expresses this?

Reply:
https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-abuse-rate-limits

Reply:
So then "GitHub treats clients ... harshly" refers to the further abuse rate limit apart from the regular rate limit?

My understanding is that the abuse rate limit is intended to limit something entirely different (too fast/concurrent calls/expensive calls vs exceeding total allowed calls per minute/hour).

Reply:
Ah sorry, it is this: 
https://docs.github.com/en/rest/guides/best-practices-for-integrators#dealing-with-rate-limits

> If you hit a rate limit, it's expected that you back off from making requests and try again later when you're permitted to do so. Failure to do so may result in the banning of your app.